connectivity/curl/include/multi_performer.h

```diff
@@ -0,0 +1,47 @@
+// Copyright Antioch. All rights reserved.
+
+#pragma once
+
+#include <curl/curl.h>
+
+#include <condition_variable>
+#include <map>
+#include <mutex>
+
+namespace antioch {
+namespace connectivity {
+namespace curl_transfer {
+
+typedef struct {
+  void (*finish_cb)(std::string);
+  size_t size;
+  char* buf;
+} EasyObj;
+
+class MultiPerformer {
+ public:
+  MultiPerformer(CURLM* multi_handle);
+  virtual ~MultiPerformer();
+
+  /**
+   * Queues the transfer. The function will return immediately, but the transfer will happen in the
+   * background. The finish_cb will be called when the transfer is finished or errors out.
+   */
+  void queue_transfer(CURL* easy_handle, void (*finish_cb)(std::string));
+
+ private:
+  void run();
+  virtual void do_transfers();
+
+  CURLM* multi_handle;
+
+  std::mutex mtx;  // synchronizes all the below objects
+  std::condition_variable cv;
+  bool flag = false;
```

Comment:
What is this a flag for

Reply:
whether there is more stuff to be transferred, lemme rename